pkg/commands/credentials.go

```diff
@@ -0,0 +1,299 @@
+	<-serverStartedChan
+
+	ex, err := os.Executable()
+	if err != nil {
+		ex = os.Args[0]
+	}
+
+	splitCmd := str.ToArgv(command)
+	cmd := exec.Command(splitCmd[0], splitCmd[1:]...)
+	cmd.Env = append(
+		os.Environ(),
+		"LAZYGIT_HOST_PORT="+hostPort,
+		"LAZYGIT_LISTENER="+currentListener,
+
+		"GIT_ASKPASS="+ex, // tell git where lazygit is located so it can ask lazygit for credentials
+
+		"LANG=en_US.UTF-8",   // Force using EN as language
+		"LC_ALL=en_US.UTF-8", // Force using EN as language
+	)
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		outString := string(out)
+		if len(outString) == 0 {
+			end <- err
+			return
+		}
+		end <- errors.New(outString)
+		return
+	}
+	end <- nil
+}
+
+// runServer starts the server that waits for events from the lazygit client
+func runServer(serverStartedChan chan struct{}, end chan error, hostPort, currentListener string) {
+	serverRunning := false
+
+	addy, err := net.ResolveTCPAddr("tcp", "127.0.0.1:"+hostPort)
+	if err != nil {
+		end <- err
+		return
+	}
+
+	inbound, err := net.ListenTCP("tcp", addy)
+	if err != nil {
+		end <- err
+		return
+	}
+
+	go func() {
+		<-end
+		if serverRunning {
+			inbound.Close()
+		}
+	}()
+
+	listener := new(Listener)
+
+	// every listener needs a different name it this is not dune rpc.RegisterName will error
+	err = rpc.RegisterName("Listener"+currentListener, listener)
+	if err != nil {
+		end <- err
+		return
+	}
+
+	serverStartedChan <- struct{}{}
+
+	serverRunning = true
+	rpc.Accept(inbound)
+	serverRunning = false
+}
+
+// GetFreePort returns a free port that can be used by lazygit
+func GetFreePort() string {
+	checkFrom := 5000
+	toReturn := ""
+	for {
+		checkFrom++
+		check := fmt.Sprintf("%v", checkFrom)
+		if IsFreePort(check) {
+			toReturn = check
+			break
+		}
+	}
+	return toReturn
+}
+
+// IsFreePort return true if the port if not in use
+func IsFreePort(port string) bool {
+	conn, err := net.Dial("tcp", "127.0.0.1:"+port)
+	if err == nil {
+		go conn.Close()
+		return false
+	}
+	return true
+}
+
+// SetupClient sets up the client
+// This will be called if lazygit is called through git
+func SetupClient() {
+	port := os.Getenv("LAZYGIT_HOST_PORT")
+	listener := os.Getenv("LAZYGIT_LISTENER")
+
+	privateKey, publicKey, err := generateKeyPair()
+	if err != nil {
+		return
+	}
+
+	client, err := rpc.Dial("tcp", "127.0.0.1:"+port)
+	if err != nil {
+		return
+	}
+
+	var data *EncryptedMessage
+	err = client.Call("Listener"+listener+".Input", InputQuestion{
+		Question:  os.Args[len(os.Args)-1],
+		Listener:  listener,
+		PublicKey: publicKey,
+	}, &data)
+	client.Close()
+	if err != nil {
+		return
+	}
+
+	msg, err := decryptMessage(privateKey, *data)
+	if err != nil {
+		return
+	}
+
+	fmt.Println(msg)
+}
+
+// HasLGAsSubProcess returns true if lazygit is a child of this process
+func HasLGAsSubProcess() bool {
+	if !ps.Supported() {
```

Comment:
I'm not familiar with this method, what does it do?

Reply:
TL;DR It's the result of making go-ps not have errors when compiling for all platforms using gox i've added placeholders for all none supporting platforms, this function simply returns false if it would use the placeholder functions that do nothing.  
The issue: https://github.com/mitchellh/go-ps/issues/28